x-pack/plugins/case/server/client/cases/utils.ts

```diff
@@ -8,127 +8,176 @@
 import { i18n } from '@kbn/i18n';
 import { flow } from 'lodash';
 import {
-  ServiceConnectorCaseParams,
-  ServiceConnectorCommentParams,
+  ActionConnector,
+  CaseResponse,
+  CaseFullExternalService,
+  CaseUserActionsResponse,
+  CommentResponse,
+  CommentResponseAlertsType,
+  CommentType,
   ConnectorMappingsAttributes,
   ConnectorTypes,
+  CommentAttributes,
+  CommentRequestUserType,
+  CommentRequestAlertType,
+} from '../../../common/api';
+import { ActionsClient } from '../../../../actions/server';
+import { externalServiceFormatters, FormatterConnectorTypes } from '../../connectors';
+import { CaseClientGetAlertsResponse } from '../../client/alerts/types';
+import { isUserContext } from '../../routes/api/utils';
+import {
+  BasicParams,
   EntityInformation,
   ExternalServiceParams,
+  ExternalServiceComment,
   Incident,
-  JiraPushToServiceApiParams,
   MapIncident,
   PipedField,
   PrepareFieldsForTransformArgs,
   PushToServiceApiParams,
-  ResilientPushToServiceApiParams,
-  ServiceNowITSMPushToServiceApiParams,
-  SimpleComment,
   Transformer,
   TransformerArgs,
   TransformFieldsArgs,
-} from '../../../../../common/api';
-import { ActionsClient } from '../../../../../../actions/server';
-export const mapIncident = async (
-  actionsClient: ActionsClient,
+} from './types';
+
+export const getLatestPushInfo = (
   connectorId: string,
-  connectorType: string,
-  mappings: ConnectorMappingsAttributes[],
-  params: ServiceConnectorCaseParams
-): Promise<MapIncident> => {
-  const { comments: caseComments, externalId } = params;
+  userActions: CaseUserActionsResponse
+): { index: number; pushedInfo: CaseFullExternalService } | null => {
+  for (const [index, action] of [...userActions].reverse().entries()) {
+    if (action.action === 'push-to-service' && action.new_value)
+      try {
+        const pushedInfo = JSON.parse(action.new_value);
+        if (pushedInfo.connector_id === connectorId) {
+          // We returned the index of the element in the userActions array.
+          // As we traverse the userActions in reverse we need to calculate the index of a normal traversal
+          return { index: userActions.length - index - 1, pushedInfo };
+        }
+      } catch (e) {
+        // Silence JSON parse errors
+      }
+  }
+
+  return null;
+};
+
+const isConnectorSupported = (connectorId: string): connectorId is FormatterConnectorTypes =>
+  Object.values(ConnectorTypes).includes(connectorId as ConnectorTypes);
+
+const getCommentContent = (comment: CommentResponse): string => {
```

Comment:
You do not have to do in your PR, but I would like us to not use comment anymore but more attachment and we can have different type/kind of attachment and we learn recently that it will be easy to rename our savedObjectType to something else.

Reply:
You are right. I followed the same wording until we enhance the code with a better naming for all comments.